keras/layers/core/dense.py

```diff
@@ -119,6 +146,54 @@ def compute_output_shape(self, input_shape):
         output_shape[-1] = self.units
         return tuple(output_shape)
 
+    def enable_lora(self, rank):
+        if self.kernel_constraint:
+            raise ValueError(
+                "Lora is incompatible with kernel constraints. "
+                "In order to enable lora on this layer, remove the "
+                "`kernel_constraint` argument."
+            )
+        if not self.built:
+            raise ValueError(
+                "Cannot enable lora on a layer that isn't yet built."
+            )
+        self._tracker.locked = False
```

Comment:
> This will basically invalidate any compiled function on a toplevel model right?

It would not. The behavior of a compiled `fit()` is that it sticks to the state of the model when it was last compiled. To get the new lorafied model, you need to recompile.